service/application/src/main/java/com/solace/maas/ep/event/management/agent/subscriber/ScanCommandMessageHandler.java

```diff
@@ -2,68 +2,31 @@
-import java.util.List;
-import java.util.UUID;
-import java.util.stream.Collectors;
-
 @Slf4j
 @Component
-@ConditionalOnProperty(name = "event-portal.gateway.messaging.standalone", havingValue = "false")
-public class ScanCommandMessageHandler extends SolaceMessageHandler<ScanCommandMessage> {
+@ConditionalOnExpression("${event-portal.gateway.messaging.standalone:false}== false && ${event-portal.managed:false} == false")
+public class ScanCommandMessageHandler extends SolaceDirectMessageHandler<ScanCommandMessage> {
 
-    private static final String DEFAULT_DESTINATION = "FILE_WRITER";
-    private final ScanManager scanManager;
+    private final ScanCommandMessageProcessor scanCommandMessageProcessor;
 
-    public ScanCommandMessageHandler(SolaceConfiguration solaceConfiguration, SolaceSubscriber solaceSubscriber, ScanManager scanManager) {
+    public ScanCommandMessageHandler(SolaceConfiguration solaceConfiguration,
+                                     SolaceSubscriber solaceSubscriber,
+                                     ScanCommandMessageProcessor scanCommandMessageProcessor) {
         super(solaceConfiguration.getTopicPrefix() + "scan/command/v1/scanStart/>", solaceSubscriber);
-        this.scanManager = scanManager;
+        this.scanCommandMessageProcessor = scanCommandMessageProcessor;
     }
 
     @Override
     public void receiveMessage(String destinationName, ScanCommandMessage message) {
         MDC.clear();
-
-        List<String> destinations = new ArrayList<>();
-        List<String> entityTypes = new ArrayList<>();
-
         log.debug("Received scan command message: {} for event broker: {}, traceId: {}",
                 message, message.getMessagingServiceId(), message.getTraceId());
-
-        message.getScanTypes().forEach(scanType -> entityTypes.add(scanType.name()));
-
-        if (message.getDestinations() == null) {
-            destinations.add(DEFAULT_DESTINATION);
-        } else {
-            message.getDestinations().forEach(destination -> destinations.add(destination.name()));
-            if (!destinations.contains(DEFAULT_DESTINATION)) {
-                destinations.add(DEFAULT_DESTINATION);
-            }
-        }
-
-        List<String> scanRequestDestinations = destinations.stream()
-                .distinct()
-                .collect(Collectors.toUnmodifiableList());
-
-        ScanRequestBO scanRequestBO = ScanRequestBO.builder()
-                .messagingServiceId(message.getMessagingServiceId())
-                .scanId(!StringUtils.isEmpty(message.getScanId()) ? message.getScanId() : UUID.randomUUID().toString())
-                .traceId(message.getTraceId())
-                .actorId(message.getActorId())
-                .scanTypes(entityTypes)
-                .destinations(scanRequestDestinations)
-                .build();
-
-        log.info("Received scan request {}. Request details: {}", scanRequestBO.getScanId(), scanRequestBO);
-
-        scanManager.scan(scanRequestBO);
```

Comment:
These common code will be used by both `directMessageHandler` and `persistentMessageHandler`. So extracted to a common class `scanCommandMessageProcessor`